packages/rule-typescript-config/src/is-valid.ts

```diff
@@ -0,0 +1,140 @@
+/**
+ * @fileoverview `typescript-config-is-valid` warns again providing an invalid typescript configuration file `tsconfig.json`.
+ */
+import * as ajv from 'ajv';
+import * as without from 'lodash.without';
+import * as groupBy from 'lodash.groupby';
+import * as map from 'lodash.map';
+
+import { Category } from 'sonarwhal/dist/src/lib/enums/category';
+import { RuleScope } from 'sonarwhal/dist/src/lib/enums/rulescope';
+import { RuleContext } from 'sonarwhal/dist/src/lib/rule-context';
+import { IRule, RuleMetadata } from 'sonarwhal/dist/src/lib/types';
+import { debug as d } from 'sonarwhal/dist/src/lib/utils/debug';
+
+import { TypeScriptConfigInvalid, TypeScriptConfigInvalidSchema } from '@sonarwhal/parser-typescript-config/dist/src/types';
+
+const debug: debug.IDebugger = d(__filename);
+
+/*
+ * ------------------------------------------------------------------------------
+ * Public
+ * ------------------------------------------------------------------------------
+ */
+
+export default class TypeScriptConfigIsValid implements IRule {
+    public static readonly meta: RuleMetadata = {
+        docs: {
+            category: Category.interoperability,
+            description: '`typescript-config-is-valid` warns again providing an invalid typescript configuration file `tsconfig.json`'
+        },
+        id: 'typescript-config/is-valid',
+        schema: [],
+        scope: RuleScope.local
+    }
+
+    public constructor(context: RuleContext) {
+        /**
+         * Returns a readable error for 'additionalProperty' errors.
+         */
+        const generateAdditionalPropertiesError = (error: ajv.ErrorObject): string => {
+            const property = error.dataPath.substr(1);
+            const additionalProperty = (error.params as ajv.AdditionalPropertiesParams).additionalProperty;
+
+            return `'${property}' ${error.message}. Additional property found '${additionalProperty}'.`;
+        };
+
+        /**
+         * Returns a readable error for 'enum' errors.
+         */
+        const generateEnumError = (error: ajv.ErrorObject): string => {
+            const property = error.dataPath.substr(1);
+            const allowedValues = (error.params as ajv.EnumParams).allowedValues;
+
+            return `'${property}' ${error.message} '${allowedValues.join(', ')}'. Value found '${error.data}'`;
+        };
+
+        /**
+         * Returns a readable error for 'pattern' errors.
+         */
+        const generatePatternError = (error: ajv.ErrorObject) => {
+            const property = error.dataPath.substr(1);
+
+            return `'${property}' ${error.message.replace(/"/g, '\'')}. Value found '${error.data}'`;
+        };
+
+        /**
+         * Returns a readable error message.
+         */
+        const prettyfy = (error: ajv.ErrorObject): string => {
+            let result: string;
+
+            switch (error.keyword) {
```

Comment:
Can we have something similar to what I said in https://github.com/sonarwhal/sonarwhal/pull/808#discussion_r171924899 ?